lib/membrane/video_compositor/temporal_scene.ex

```diff
@@ -1,4 +1,4 @@
-defmodule Membrane.VideoCompositor.Scene.Temporal do
+defmodule Membrane.VideoCompositor.TemporalScene do
```

Comment:
I don't like the proposed structure mainly, because when user alias it (with such long names, they probably will), they become unreadable.

Aliasing `Membrane.VideoCompositor.Scene.Temporal` means, that now all user `typespecs` use this weird `Tempotal.t()` type, that is hardly resembling `scene` hiding underneath it.

Using `Membrane.VideoCompositor.Temporal.Scene` means, that to avoid ambiguity, we have to `alias` with `as` one of: 
- `scene`
- `temporal scene`

(BTW while writing such code, I would probably alias it as `TemporalScene` :) )

Since `temporal scene` isn't a struct, we can stick with `Membrane.VideoCompositor.Scene.Temporal` if you want to (it's only a hindrance in `typespecs`, with the possibility of solving that using `Scene.Temporal.t()` in them).

Aside from that, I think that `temporal scene` is rather an alternative to `scene`, than its extension. Therefore, putting this module into `Membrane.VideoCompositor.TemporalScene` doesn't sound wrong to me.
 
Let me know what you think about it. It's up to you to decide whether we want `Membrane.VideoCompositor.TemporalScene` or `Membrane.VideoCompositor.Scene.Temporal`, I don't have a strong opinion on that. I just wanted to provide arguments for `Membrane.VideoCompositor.TemporalScene`, but if you still think `Membrane.VideoCompositor.Scene.Temporal` is better, we can use that.

Reply:
After some discussion and consulting with @bblaszkow06 , this is what was suggested and what I think:

- There are no meaningful reasons for `VideoCompositor.TemporalScene` to be a separate module
-  Types in `VideoCompositor.TemporalScene` can be moved to `VideoCompositor.Handler` - with `VideoCompositor.TemporalScene.t()` moved to `VideoCompositor.Handler.temporal_scene()`
- `##Expiration` part of moduledoc should be moved to typedoc of `temporal_scene()`
- `##Examples` part of moduledoc should be moved to moduledoc of `Handler`